### PR TITLE
Add Mock sniff.

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/spryker/code-sniffer/issues/204

### CODE REVIEW

File: Spryker/Sniffs/Testing/MockSniff.php [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        if (!$returnTypeHint || $returnTypeHint->getTypeHint() !== 'MockObject') {
+            return;
+        }
+
+        $fix = $phpcsFile->addFixableError('MockObject must not be typehinted.', $returnTypeHint->getStartPointer(), 'InvalidTypehint');
+        if (!$fix) {
+            return;
+        }
+
+        $this->removeReturnTypeHint($phpcsFile, $stackPtr, $returnTypeHint);
```
Comment (Incognito) [inline Spryker/Sniffs/Testing/MockSniff.php]: Consider moving this call into `process`, it removes a frame in the call stack and leads to easier debuging, and learnability for future developers.

Comment (dereuromark) [reply Spryker/Sniffs/Testing/MockSniff.php]: Originally, this was used more than once.

File: Spryker/Sniffs/Testing/MockSniff.php [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    {
+        $colonPointer = $phpcsFile->findPrevious(T_COLON, $returnTypeHint->getStartPointer(), $stackPtr);
+        if (!$colonPointer) {
+            return;
+        }
+
+        $phpcsFile->fixer->beginChangeset();
+        $endPtr = $returnTypeHint->getEndPointer();
+
+        for ($i = $colonPointer; $i <= $endPtr; $i++) {
```
Comment (Incognito) [inline Spryker/Sniffs/Testing/MockSniff.php]: Consider renaming `colonPointer` above to `pointerToPreviousColon`

```suggestion
        for ($colonPointer = $pointerToPreviousColon; 
```

File: Spryker/Sniffs/Testing/MockSniff.php [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+    /**
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile
+     * @param int $stackPtr
+     * @param string[] $docBlockReturnTypes
+     * @param \SlevomatCodingStandard\Helpers\ReturnTypeHint|null $returnTypeHint
+     *
+     * @return void
+     */
+    protected function assertDocBlockReturnAnnotation(File $phpcsFile, int $stackPtr, array $docBlockReturnTypes, ?ReturnTypeHint $returnTypeHint): void
```
Comment (Incognito) [inline Spryker/Sniffs/Testing/MockSniff.php]: Can the contents of the method be broken up into smaller methods with meaningful names? The parsing and chain of if-statements looks a bit intimidating.

Comment (dereuromark) [reply Spryker/Sniffs/Testing/MockSniff.php]: Probably not without heavy refactoring and/or duplication of stuff at costs of runtime. As such I think this has to make due.

Comment (Incognito) [reply Spryker/Sniffs/Testing/MockSniff.php]: If the if-blocks don't share state it might be a collection of classes with the same method interface.

**Pull request merged**